aries_cloudagent/protocols/issue_credential/v2_0/formats/vc_di/handler.py

```diff
@@ -0,0 +1,458 @@
+                "Missing nonce in credential offer with anoncreds link secret binding method"
+            )
+
+        nonce = cred_offer["binding_method"]["anoncreds_link_secret"]["nonce"]
+        cred_def_id = cred_offer["binding_method"]["anoncreds_link_secret"][
+            "cred_def_id"
+        ]
+
+        async def _create():
+            anoncreds_registry = self.profile.inject(AnonCredsRegistry)
+
+            cred_def_result = await anoncreds_registry.get_credential_definition(
+                self.profile, cred_def_id
+            )
+
+            holder = AnonCredsHolder(self.profile)
+            request_json, metadata_json = await holder.create_credential_request(
+                cred_offer, cred_def_result.credential_definition, holder_did
+            )
+
+            return {
+                "request": json.loads(request_json),
+                "metadata": json.loads(metadata_json),
+            }
+
+        cache_key = f"credential_request::{cred_def_id}::{holder_did}::{nonce}"
+        cred_req_result = None
+        cache = self.profile.inject_or(BaseCache)
+        if cache:
+            async with cache.acquire(cache_key) as entry:
+                if entry.result:
+                    cred_req_result = entry.result
+                else:
+                    cred_req_result = await _create()
+                    await entry.set_result(cred_req_result, 3600)
+        if not cred_req_result:
+            cred_req_result = await _create()
+
+        detail_record = V20CredExRecordVCDI(
+            cred_ex_id=cred_ex_record.cred_ex_id,
+            cred_request_metadata=cred_req_result["metadata"],
+        )
+
+        async with self.profile.session() as session:
+            await detail_record.save(session, reason="create v2.0 credential request")
+
+        return self.get_format_data(CRED_20_REQUEST, cred_req_result["request"])
+
+    async def receive_request(
+        self, cred_ex_record: V20CredExRecord, cred_request_message: V20CredRequest
+    ) -> None:
+        """Receive vcdi credential request."""
+        if not cred_ex_record.cred_offer:
+            raise V20CredFormatError(
+                "vcdi issue credential format cannot start from credential request"
+            )
+
+    async def issue_credential(
+        self, cred_ex_record: V20CredExRecord, retries: int = 5
+    ) -> CredFormatAttachment:
+        """Issue vcdi credential."""
+        await self._check_uniqueness(cred_ex_record.cred_ex_id)
+
+        attached_credential = cred_ex_record.cred_offer.attachment(
+            VCDICredFormatHandler.format
+        )
+        detail_credential = VCDICredAbstract.deserialize(attached_credential)
+        binding_proof = cred_ex_record.cred_request.attachment(
+            VCDICredFormatHandler.format
+        )
+        detail_proof = VCDICredRequest.deserialize(binding_proof)
+        manager = self.profile.inject(VcLdpManager)
+        # TODO - implement a separate create_credential for vcdi
```

Comment:
Does this TODO need to be addressed before the PR is merged?

Reply:
yes we are trying to see how exactly to do it...agree this is the key place.  @dhh1128 or @ianco any suggestion on the right method very welcome.